lifecycle-operator/test/component/appversion/appversion_suite_test.go

```diff
@@ -34,23 +35,36 @@ var (
 	spanRecorder *sdktest.SpanRecorder
 )
 
-const KeptnNamespace = "keptnlifecycle"
+const (
+	KeptnNamespace     = "keptnlifecycle"
+	traceComponentName = "keptn/lifecycle-operator/appversion"
+)
 
 var _ = BeforeSuite(func() {
 	var readyToStart chan struct{}
 	ctx, k8sManager, tracer, spanRecorder, k8sClient, readyToStart = common.InitSuite()
 
+	TracerFactory := &common.TracerFactory{Tracer: tracer}
+	EvaluationHandler := evaluation.NewEvaluationHandler(
+		k8sManager.GetClient(),
+		controllercommon.NewK8sSender(k8sManager.GetEventRecorderFor("test-appversion-controller")),
+		GinkgoLogr,
+		TracerFactory.GetTracer(traceComponentName),
+		k8sManager.GetScheme(),
+		&telemetry.SpanHandler{})
+
 	config.Instance().SetDefaultNamespace(KeptnNamespace)
 
 	// //setup controllers here
 	controller := &keptnappversion.KeptnAppVersionReconciler{
-		Client:        k8sManager.GetClient(),
-		Scheme:        k8sManager.GetScheme(),
-		EventSender:   controllercommon.NewK8sSender(k8sManager.GetEventRecorderFor("test-appversion-controller")),
-		Log:           GinkgoLogr,
-		Meters:        common.InitKeptnMeters(),
-		SpanHandler:   &telemetry.SpanHandler{},
-		TracerFactory: &common.TracerFactory{Tracer: tracer},
+		Client:            k8sManager.GetClient(),
+		Scheme:            k8sManager.GetScheme(),
+		EventSender:       controllercommon.NewK8sSender(k8sManager.GetEventRecorderFor("test-appversion-controller")),
+		Log:               GinkgoLogr,
+		Meters:            common.InitKeptnMeters(),
+		SpanHandler:       &telemetry.SpanHandler{},
+		TracerFactory:     &common.TracerFactory{Tracer: tracer},
```

Comment:
also let's use `tracerFactory` here

Reply:
For the key?

Reply:
no, you already created the factory on line 47, there is no need to create another one

Reply:
Hi @odubajDT I can't rename `TracerFactory` to `tracerFactory` as it was defined as TracerFactory` in the controller [here](https://github.com/keptn/lifecycle-toolkit/blob/main/lifecycle-operator/controllers/lifecycle/keptnappversion/controller.go#L52)

Reply:
No I meant something else. Instead of creating a new factory with `common.TracerFactory` you can use the already existing one (you defined in on line 47 of this file)

Reply:
Alright. it makes a lot of sense now. 
Thanks